internal/dependency/dependency.go

```diff
@@ -118,15 +121,36 @@ func (q *QueryOptions) Merge(o *QueryOptions) *QueryOptions {
 	return &r
 }
 
+func (q *QueryOptions) Context() context.Context {
```

Comment:
was curious to learn more about this method. i don't see this called in this pr. is the intention for the client to use it? also a little curious about `return context.Background()` vs. returning nil?

Reply:
Very true, it's not used. I can omit it. I had copied this from the consul API client https://github.com/hashicorp/consul/blob/master/api/api.go#L171-L185

Not being the original author, but I do like how `context.Background()` is returned because most functions that take in `context` require it to not be nil and would panic.

Reply:
Nice catch. I vote for removing unused code.

Reply:
ahh ok. i wasn't really sure about returning a new context that's not set on QueryOptions and hadn't come across something like that before. cool. thanks for sharing the context!